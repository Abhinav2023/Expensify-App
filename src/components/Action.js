import React from "react"

const Action=(props)=>(
    <div>
        <button className="big-button" disabled={!props.hasOption} onClick={props.RandomPick}>What Should I do?</button>
    </div>
)
export default Action;