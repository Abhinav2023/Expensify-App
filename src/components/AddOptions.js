import React from "react";

export default class AddOptions extends React.Component {
    state={
        error: undefined
    };
    addOptions =(e)=> {
        e.preventDefault();
        const option = e.target.elements.option.value.trim();
        const error= this.props.AddOption(option);
        this.setState(() => ({ error }));
  
        if (!error) {
            e.target.elements.option.value = '';
        }
    }
    render() {
        return (
            <div>
                {this.state.error && <p className="add-option-error">{this.state.error}</p>}
                <form className="add-option" onSubmit={this.addOptions}>
                    <input className="add-option__input"type="text" name="option" />
                    <button className="button">Add Option</button>
                </form>
            </div>
        )
    }
}