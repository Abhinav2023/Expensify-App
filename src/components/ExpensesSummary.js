import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expense-total';

export const ExpensesSummary=({expensesCount,expensesTotal})=>{
    const expenseWord=expensesCount===1 ? 'expense' : 'expenses';
    const formattedExpensesTotal=numeral(expensesTotal/100).format('$0,0.00');
    return(
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                    <div class="button_cont" align="center">
                        <a class="example_e" href="/create">
                            Add Expense
                        </a>
                    </div>
                </div>
            </div>
        </div>
    )
};

const mapStateToProps=(state)=>{
    const visibleExpenses=selectExpenses(state.expenses,state.filters);
    return{
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);