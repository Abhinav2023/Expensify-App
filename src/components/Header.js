import React from "react";

const Header=(props)=>(
    <div className="header">
        <div className="container">
            <h1 className="header__title">{props.title}</h1>
            {props.subtitle && <h2 className="header__subtitle">{props.subtitle}</h2>}
        </div>
    </div>
)
Header.defaultProps={
    title: "Indecision",
    subtitle: "Put Your Hands in the control of keyboard"
}
export default Header;