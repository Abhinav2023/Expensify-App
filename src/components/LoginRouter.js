import React from 'react';
import {connect } from 'react-redux';
import {startLogin} from '../actions/auth';
export const LoginPage=({startLogin})=>(
    <div>
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify App</h1>
                <p>This is my first expensify app</p>
                <div class="button_cont" align="center">
                    <a onClick={startLogin} class="example_d" >Login With Google</a>
                </div>
            </div>
        </div>
    </div>
);

const mapDispatchProps=(dispatch)=>({
    startLogin: ()=>dispatch(startLogin())
})

export default connect(undefined, mapDispatchProps)(LoginPage);