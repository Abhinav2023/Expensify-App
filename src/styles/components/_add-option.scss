.add-option-error{
    color: $offwhite;
    font-style: italic;
    margin: $m-base 0 0 0;
    padding: 0 $m-base;
}

.add-option{
    display: flex;
    padding: $m-base;
}

.add-option__input{
    flex-grow: 1;
    background: $dark-blue;
    border: none;
    color: $offwhite;
    padding: $s-size;
    margin-right: $s-size;
    border-bottom: .3rem solid darken($dark-blue, 10%)
}