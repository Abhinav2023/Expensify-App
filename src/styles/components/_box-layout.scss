.box-layout{
    align-items: center;
    background: url(' /images/bg.jpg');
    max-width:100%;
    height:100vh;
    width: 1000vw;
    background-size: cover;
    display: flex;
    justify-content: center;
}

.box-layout__box{
    background: fade-out($color: white, $amount: .15);
    width: 25rem;
    padding: $l-size $m-size;
    border-radius: 3px;
    text-align: center;
}

.box-layout__title{
    line-height: 1;
    margin: 0 0 $m-size 0;
}