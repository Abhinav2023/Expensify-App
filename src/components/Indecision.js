import React from "react";
import AddOptions from "./AddOptions";
import Options from "./Options";
import Action from "./Action";
import Header from "./Header";
import OptionModal from "./OptionModal";

export default class Indecision extends React.Component {
    state={
        options: [],
        selectedOption: undefined
    }

    RandomPick=()=>{
        const randomNum = Math.floor(Math.random() * this.state.options.length);
        const option = this.state.options[randomNum];
        this.setState(()=>({
            selectedOption: option
        }))
    }

    RemoveAll=()=>{
        this.setState(()=>({options: []}))
    }

    RemoveOption=(optionToRemove)=>{
        this.setState((prevState)=>({
            options: prevState.options.filter((option)=>optionToRemove!==option)
        }));
        
    }

    handleClearSelection=()=>{
        this.setState(()=>({
            selectedOption:undefined
        }))
    }

    AddOption=(option)=>{
        if(!option){
            return "Enter Valid entry to add item"
        }else if(this.state.options.indexOf(option)>-1){
            return "This already exist";
        }
        this.setState((prevState)=>({options: prevState.options.concat(option)}))
    }

    componentDidMount(){
        try {
            const json = localStorage.getItem('options');
            const options = JSON.parse(json);
            if (options) {
              this.setState(() => ({ options }));
            }
        } catch (e) {
        // proton bsdk
        }
    }
    componentDidUpdate(prevProps,prevState){
        if (prevState.options.length !== this.state.options.length) {
            const json = JSON.stringify(this.state.options);
            localStorage.setItem('options', json);
          }
    }
    componentWillUnmount(){
        console.log("Unmounting the component")
    }

    

    render() {
        const title = "Indecision";
        const subtitle = "Put Your Life in the hands of computer";
        return (
            <div>
                <Header />
                <div className="background">
                    <div className="container">
                        <Action RandomPick={this.RandomPick} hasOption={this.state.options.length >0}/>
                        <div className="widget">
                            <Options options={this.state.options} RemoveAll={this.RemoveAll} RemoveOption={this.RemoveOption}/>
                            <AddOptions AddOption={this.AddOption}/>
                        </div>
                    </div>
                    <OptionModal
                        selectedOption={this.state.selectedOption}
                        handleClearSelection={this.handleClearSelection}
                    />
                </div>
                
            </div>
        );
    }
}