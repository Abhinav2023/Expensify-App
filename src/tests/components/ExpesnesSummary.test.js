import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test("should correctly render ExpensSummary with 1 expense",()=>{
    const wrapper =shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpensSummary with 1 expense",()=>{
    const wrapper =shallow(<ExpensesSummary expensesCount={32} expensesTotal={2353214134} />);
    expect(wrapper).toMatchSnapshot();
});