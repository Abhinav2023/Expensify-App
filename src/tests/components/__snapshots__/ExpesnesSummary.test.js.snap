// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render ExpensSummary with 1 expense 2`] = `
<div>
  <h1>
    Viewing 
    32
     
    expenses
     totalling 
    $23,532,141.34
  </h1>
</div>
`;
